trusted_shuffler/trusted_shuffler/src/lib.rs

```diff
@@ -33,45 +33,76 @@ use std::{
 use tokio::{sync::oneshot, time::Duration};
 
 #[derive(Debug, PartialEq)]
-pub struct TrustedShufflerRequest {
+pub struct SecretRequest {
     pub body: Vec<u8>,
     pub headers: hyper::HeaderMap,
     pub uri: Uri,
 }
 
-impl TrustedShufflerRequest {
-    // We only compare the body, and we need revisit this and decide if this
-    // is how we want to sort our requests.
-    fn compare_body(&self, r2: &TrustedShufflerRequest) -> Ordering {
+// TODO(mschett): Add real stuff!
+impl SecretRequest {
+    fn decrypt(self) -> PlainRequest {
+        PlainRequest {
+            body: self.body,
+            headers: self.headers,
+            uri: self.uri,
+        }
+    }
+}
+
+pub struct PlainRequest {
+    pub body: Vec<u8>,
```

Comment:
Is it intentional that this struct has `body`, but `SecretResponse` has `data`? It seems they should use the same terminology (and also types, if possible?)

Reply:
Yes, but I'd do that in a seperate #3282 .